tests/test_whitelist.py

```diff
@@ -27,11 +28,814 @@ def test_load_org_asn(org, asn, mock_db):
+            {"example.com": {"from": "dst", "what_to_ignore": "both"}}
+        ),
+        "organizations": json.dumps(
+            {"google": {"from": "both", "what_to_ignore": "both"}}
+        ),
+        "mac": json.dumps(
+            {"b1:b1:b1:c1:c2:c3": {"from": "src", "what_to_ignore": "alerts"}}
+        ),
+    }
+    all_whitelist = whitelist.get_all_whitelist()
+    assert all_whitelist is not None
+    assert "IPs" in all_whitelist
+    assert "domains" in all_whitelist
+    assert "organizations" in all_whitelist
+    assert "mac" in all_whitelist
+
+
+@pytest.mark.parametrize(
+    "flow_data, whitelist_data, expected_result",
+    [
+        (  # testing_is_whitelisted_flow_with_whitelisted_organization_but_ip_or_domain_not_whitelisted
+            MagicMock(
+                saddr="9.8.7.6", daddr="5.6.7.8", type_="http", host="org.com"
+            ),
+            {
+                "organizations": {
+                    "org": {"from": "both", "what_to_ignore": "flows"}
+                }
+            },
+            False,
+        ),
+        (  # testing_is_whitelisted_flow_with_non_whitelisted_organization_but_ip_or_domain_whitelisted
+            MagicMock(
+                saddr="1.2.3.4",
+                daddr="5.6.7.8",
+                type_="http",
+                host="whitelisted.com",
+            ),
+            {"IPs": {"1.2.3.4": {"from": "src", "what_to_ignore": "flows"}}},
+            False,
+        ),
+        (  # testing_is_whitelisted_flow_with_whitelisted_source_ip
+            MagicMock(
+                saddr="1.2.3.4",
+                daddr="5.6.7.8",
+                type_="http",
+                server_name="example.com",
+            ),
+            {"IPs": {"1.2.3.4": {"from": "src", "what_to_ignore": "flows"}}},
+            False,
+        ),
+        (  # testing_is_whitelisted_flow_with_both_source_and_destination_ips_whitelisted
+            MagicMock(saddr="1.2.3.4", daddr="5.6.7.8", type_="http"),
+            {
+                "IPs": {
+                    "1.2.3.4": {"from": "src", "what_to_ignore": "flows"},
+                    "5.6.7.8": {"from": "dst", "what_to_ignore": "flows"},
+                }
+            },
+            False,
+        ),
+        (
+            # testing_is_whitelisted_flow_with_whitelisted_mac_address_but_ip_not_whitelisted
+            MagicMock(
+                saddr="9.8.7.6",
+                daddr="1.2.3.4",
+                smac="b1:b1:b1:c1:c2:c3",
+                dmac="a1:a2:a3:a4:a5:a6",
+                type_="http",
+                server_name="example.org",
+            ),
+            {
+                "mac": {
+                    "b1:b1:b1:c1:c2:c3": {
+                        "from": "src",
+                        "what_to_ignore": "flows",
+                    }
+                }
+            },
+            False,
+        ),
+    ],
+)
+def test_is_whitelisted_flow(
+    mock_db, flow_data, whitelist_data, expected_result
```

Comment:
i will split  is_whitelisted_flow() into smaller functions and will let you know so we can better test it, it's very long now and this test case is testing too many different things at once